src/redux/reducers/searchText.js

```diff
@@ -0,0 +1,198 @@
+import { handleActions } from 'redux-actions';
+import actionTypes from '../actionTypes';
+import { searchMessages } from '../../utils/searchMessages';
+
+const defaultState = {
+  search: false,
+  searchState: {
+    markedMessagesByID: {},
+    markedIDs: [],
+    markedIndices: [],
+    scrollLimit: 0,
+    scrollIndex: -1,
+    scrollID: null,
+    caseSensitive: false,
+    searchIndex: 0,
+    searchText: '',
+  },
+};
+
+export default handleActions(
+  {
+    [actionTypes.SEARCH_TEXT_CHANGED](state, { payload }) {
+      // TODO: Call close snackbar from Top Bar
+      const { messages, messagesByID, searchText } = payload;
+      let { searchState } = state;
+      // const searchText = event.target.value;
+      if (searchText.trim()) {
+        const markingData = searchMessages(
+          messages,
+          messagesByID,
+          searchText,
+          searchState.caseSensitive,
+        );
+        const { markedMessagesByID, markedIDs, markedIndices } = markingData;
+        searchState = {
+          markedMessagesByID: markedMessagesByID,
+          markedIDs: markedIDs,
+          markedIndices: markedIndices,
+          scrollLimit: markedIDs.length,
+          scrollIndex: 0,
+          scrollID: markedIDs[0],
+          caseSensitive: searchState.caseSensitive,
+          searchIndex: 1,
+          searchText: searchText,
+        };
+        if (markedIDs.length === 0 && searchText.trim()) {
+          // if no Messages are marked(i.e no result) and the search query is not empty
```

Comment:
This functionality can be achieved by adding a new state to searchState

Reply:
@akshatnitd I'll have to think of some creative way to optimize searchMessage Util, maybe then I can get rid of redundant variables in this reducer. It is currently dependent on what the util is returning. So, I'll rater open a new issue for searchMessage util optimization.